MetadataMigration/src/test/java/org/opensearch/migrations/KnnSettingsTransformationTest.java

```diff
@@ -0,0 +1,112 @@
+package org.opensearch.migrations;
+
+import java.io.File;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.opensearch.migrations.bulkload.framework.SearchClusterContainer;
+import org.opensearch.migrations.commands.MigrationItemResult;
+import org.opensearch.migrations.snapshot.creation.tracing.SnapshotTestContext;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+/**
+ * Test to reproduce the KNN settings uniqueness when migrating from OpenSearch 1.3 to OpenSearch 2.x.
+ */
+@Tag("isolatedTest")
+@Slf4j
+class KnnSettingsTransformationTest extends BaseMigrationTest {
+    @TempDir
+    protected File localDirectory;
+
+    private static Stream<Arguments> scenarios() {
+        var source = SearchClusterContainer.OS_V1_3_16;
+        var target = SearchClusterContainer.OS_V2_19_1;
+        return Stream.of(Arguments.of(source, target));
+    }
+
+    @ParameterizedTest(name = "KNN Settings Transformation From {0} to {1}")
+    @MethodSource(value = "scenarios")
+    void knnSettingsTransformationTest(
+            SearchClusterContainer.ContainerVersion sourceVersion,
+            SearchClusterContainer.ContainerVersion targetVersion) {
+        try (
+                final var sourceCluster = new SearchClusterContainer(sourceVersion);
+                final var targetCluster = new SearchClusterContainer(targetVersion)
+        ) {
+            this.sourceCluster = sourceCluster;
+            this.targetCluster = targetCluster;
+            performKnnSettingsTest();
+        }
+    }
+
+    @SneakyThrows
+    private void performKnnSettingsTest() {
+        startClusters();
+
+        String flatDottedFormat = "{\n" +
+                "  \"settings\": {\n" +
+                "    \"index\": {\n" +
+                "      \"number_of_shards\": \"1\",\n" +
+                "      \"knn.algo_param.ef_search\": 100,\n" +
+                "      \"knn\": \"true\",\n" +
```

Comment:
Can you set up this test to run twice, with these two lines in different orders.  I could see a refactoring of the code that does one-pass instead of two and not doing the right thing